### DISCUSSION TITLE
Working with files

### DISCUSSION THREAD

**Discussion by donkey-donkey** [tauri-apps/tauri#1579] [Q&A] [answered]
Hi All,
We are really loving the tauri framework. our app runs super smooth on macOS and linux. 
We are wondering how do we work with opening files and saving files in a single window app or CLI app?
Our app is not running node. We have a react based app that would like to open a local file and save a file locally. Not sure how to do this. Is there a local files api for tauri apps?

thanks

**Comment by nothingismagick** [comment]
You have to be a bit more specific. Please edit your question to explicitly state what you mean with "work with files". Because for some of us, this is still very true:

https://en.wikipedia.org/wiki/Everything_is_a_file

**Reply by donkey-donkey** [reply]
updated. thanks.

**Comment by nothingismagick** [comment, marked as answer]
Yes there is a Prompt api for choosing a file to open - and also a Save As:
https://tauri.studio/en/docs/api/js/modules/dialog

There is also a js api for filesystem work:
https://tauri.studio/en/docs/api/js/modules/fs

And finally there is also the path api:
https://tauri.studio/en/docs/api/js/modules/path

**Reply by donkey-donkey** [reply]
you all rock! thanks so much

**Reply by nothingismagick** [reply]
If that answered your question, please feel free to mark as answered.

**Reply by dchicchon** [reply]
Instead of using the js system, is it possible to save utilizing the tauri api? Haven't anything that is similar to the answered method

**Reply by FabianLars** [reply]
@dchicchon Since you didn't react to the [discord message](https://discord.com/channels/616186924390023171/990656096807047228/990657482173067344) i'm gonna post the same answer here too.
You have access to everything rust has to offer, including https://doc.rust-lang.org/std/fs/struct.File.html.
So writing a file can be as easy as:
```rs
use std::{fs::File, io::Write};
let file = File::create(path)?;
file.write_all(content)?;
```

**Reply by oshibka404** [reply]
Updated links for everyone in 2025:

Prompt API for choosing a file to open - and also a Save As:
https://v2.tauri.app/plugin/dialog/ (available as a plugin)

Filesystem API
https://v2.tauri.app/plugin/file-system/ (available as a plugin)

Path API
https://v2.tauri.app/reference/javascript/api/namespacepath/

**Comment by Shivang-idrive** [comment]
I want to check if a particular  file exists or not. Anyone can help?

**Reply by FabianLars** [reply]
Rust: https://doc.rust-lang.org/std/path/struct.Path.html#method.exists
JS: https://tauri.app/v1/api/js/fs#exists

**Reply by jensim** [reply]
Im confused by this and ended up here when googling for "tauri fs exist boolean" 🤷😅
On the JS side, you get a `Promise<void>`, this feels odd to me, as the question of whether or not something exists naturally should be answered with yes or no, (boolean). A nonexistent file if asked, is not an exceptional state, is it?

**Reply by FabianLars** [reply]
It's a bug. It returns a boolean, i just messed up the type annotations
